kubernetes_env/benchmark/benchmark.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+import argparse
+import logging
+import subprocess
+import os
+import sys
+import signal
+import requests
+import time
+import matplotlib.pyplot as plt
+import json
+from pathlib import Path
+
+DIRS = Path(__file__).resolve().parents
+sys.path.append(str(DIRS[1]))
+import kube_env
+import kube_util as util
+
+
+log = logging.getLogger(__name__)
+FILE_DIR = DIRS[0]
+FILTER_DIR = FILE_DIR.joinpath("cpp_filter")
+GRAPHS_DIR = FILE_DIR.joinpath("graphs")
+DATA_DIR = FILE_DIR.joinpath("data")
+FORTIO_DIR = DIRS[2].joinpath("bin/fortio")
+
+
+def plot(xaxis_data, yaxis_data, xaxis_label, yaxis_label, query, graph_name):
+    plt.plot(xaxis_data, yaxis_data, marker='o')
+    plt.xlabel(xaxis_label)
+    plt.ylabel(yaxis_label)
+    plt.title(query)
+    util.check_dir(GRAPHS_DIR)
+    plt.savefig(str(GRAPHS_DIR.joinpath(f"{graph_name}.png")))
+
+
+def run_fortio(platform, threads, qps, run_time):
+    util.check_dir(DATA_DIR)
+    _, _, gateway_url = kube_env.get_gateway_info(platform)
```

Comment:
I would move this out of this function and after check_kubernetes_status since the gateway will be constant from that point on.